lib/src/view/opening_explorer/opening_explorer_screen.dart

```diff
@@ -0,0 +1,845 @@
+                                ),
+                              ],
+                            ),
+                          ),
+                          Flexible(
+                            fit: FlexFit.loose,
+                            child: Column(
+                              mainAxisAlignment: MainAxisAlignment.start,
+                              children: [
+                                Expanded(
+                                  child: PlatformCard(
+                                    margin: const EdgeInsets.all(
+                                      kTabletBoardTableSidePadding,
+                                    ),
+                                    semanticContainer: false,
+                                    child: _OpeningExplorer(
+                                      pgn: pgn,
+                                      options: options,
+                                    ),
+                                  ),
+                                ),
+                              ],
+                            ),
+                          ),
+                        ],
+                      )
+                    : Column(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: [
+                          if (isTablet)
+                            Padding(
+                              padding: const EdgeInsets.all(
+                                kTabletBoardTableSidePadding,
+                              ),
+                              child: AnalysisBoard(
+                                pgn,
+                                options,
+                                boardSize,
+                                isTablet: isTablet,
+                              ),
+                            )
+                          else
+                            AnalysisBoard(
+                              pgn,
+                              options,
+                              boardSize,
+                              isTablet: isTablet,
+                            ),
+                          Expanded(
+                            child: _OpeningExplorer(pgn: pgn, options: options),
+                          ),
+                        ],
+                      );
+              },
+            ),
+          ),
+        ),
+        _BottomBar(pgn: pgn, options: options),
+      ],
+    );
+  }
+}
+
+class _OpeningExplorer extends ConsumerStatefulWidget {
+  const _OpeningExplorer({
+    required this.pgn,
+    required this.options,
+  });
+
+  final String pgn;
+  final AnalysisOptions options;
+
+  @override
+  ConsumerState<_OpeningExplorer> createState() => _OpeningExplorerState();
+}
+
+class _OpeningExplorerState extends ConsumerState<_OpeningExplorer> {
+  final Map<OpeningExplorerCacheKey, OpeningExplorerEntry> cache = {};
+
+  @override
+  Widget build(BuildContext context) {
+    final ctrlProvider =
```

Comment:
Since this is assigned with the return of `ref.watch` the name is not good here. What we have is the state of analysisController. So `analysisState` would be the proper name.